container/go/pkg/compat/index.go

```diff
@@ -0,0 +1,198 @@
+/// Copyright 2015 The Bazel Authors. All rights reserved.
```

Comment:
I don't think we need index.go or have enough information to create a index object, as we are not really getting a ImageIndex from the legacy format. We are just getting an image.

Reply:
You are trying to use the same oci.Read function in pusher, which takes an path to an ImageIndex directory. So you implement the whole index.go here. But it doesn't apply to this case. We should not call the same oci.Read function in pusher. But instead, write a new Read function in this compat package which takes a path to the directory that contains the legacy image, and directly read a v1.Image out of it.

Reply:
And that read function is used in pusher.go when the format is legacy.